src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,148 @@
+
+      if (person.motherName) {
+        result = person.motherName
+          .toLocaleLowerCase()
+          .includes(query.toLocaleLowerCase());
+      }
+
+      if (person.fatherName && !result) {
+        result = person.fatherName
+          .toLocaleLowerCase()
+          .includes(query.toLocaleLowerCase());
+      }
+
+      return result;
+    });
+  }
+
+  if (centuries.length) {
+    resultData = resultData.filter(person => {
+      return centuries.includes(Math.ceil(person.born / 100).toString());
+    });
+  }
+
+  if (sort) {
+    resultData = resultData.sort((person1, person2) => {
+      switch (sort) {
+        case 'name':
+        case 'sex':
+          return person1[sort].localeCompare(person2[sort]);
+
+        case 'born':
+        case 'died':
+          return person1[sort] - person2[sort];
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (order === 'desc') {
+    resultData.reverse();
+  }
+
+  return resultData;
+}
 
 export const PeoplePage = () => {
+  const [data, setData] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+  const [searchParams] = useSearchParams();
+
+  const currentQuery = searchParams.get('query');
+  const currentFilter = searchParams.get('sex') || '';
+  const selectedCenturies = searchParams.getAll('centuries');
+  const sortBy = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+  const visiblePersonList = getFilteredPersonList(
+    data,
+    currentQuery,
+    currentFilter,
+    selectedCenturies,
+    sortBy,
+    order,
+  );
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(response => setData(getExtendedPersonList(response)))
+      .catch(() => setHasError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
   return (
     <>
       <h1 className="title">People Page</h1>
-
       <div className="block">
         <div className="columns is-desktop is-flex-direction-row-reverse">
           <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
+            {!isLoading && <PeopleFilters />}
           </div>
 
           <div className="column">
             <div className="box table-container">
-              <Loader />
-
-              <p data-cy="peopleLoadingError">Something went wrong</p>
-
-              <p data-cy="noPeopleMessage">There are no people on the server</p>
+              {isLoading ? (
+                <Loader></Loader>
+              ) : (
+                <>
+                  {hasError && (
+                    <p data-cy="peopleLoadingError" className="has-text-danger">
+                      Something went wrong
+                    </p>
+                  )}
 
-              <p>There are no people matching the current search criteria</p>
+                  {!data.length && !hasError && (
+                    <p data-cy="noPeopleMessage">
```

Comment:
Would be great to create named consts for better understanding, smth like this
```suggestion
                  {isEmptyList && (
                    <p data-cy="noPeopleMessage">
```